src/Core/Domain/Module/Command/BulkToggleModuleStatusCommand.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Module\Command;
+
+use Module;
+
+/**
+ * Bulk toggles module status
+ */
+class BulkToggleModuleStatusCommand
```

Comment:
I'm sorry I think this Command is wrong.

A Command is a pure data object. It contains the data needed for the system to perform the action. So it should contain module identifiers or names, not module instances.

A Command should be serializable, we should be able to put it in a queue system. It should be serializable into JSON. It's a message. It should not contain the very items that could be modified.

You need to load the modules inside the Handler, not before you dispatch the Command. The system must do the job of fetching and updating data, not the code that calls it.

Reply:
@matks I understand. It's fixed.